allensdk/brain_observatory/behavior/running_processing.py

```diff
@@ -14,40 +16,378 @@ def calc_deriv(x, time):
+    if isinstance(fill_value, float):
+        # Circumvent issue if int-like array with np.nan as fill
+        shifted = np.roll(arr, periods).astype(float)
+    else:
+        shifted = np.roll(arr, periods)
+    shifted[:periods] = fill_value
+    return shifted
+
+
+def deg_to_dist(angular_speed: np.ndarray) -> np.ndarray:
+    """
+    Takes the angular speed (radians/s) at each step in radians, and
+    computes the linear speed in cm/s.
+
+    Parameters
+    ----------
+    angular_speed: np.ndarray (1d)
+        1d array of angular speed in radians/s
+    Returns
+    -------
+    np.ndarray (1d)
+        Linear speed in cm/s at each time point.
+    """
     wheel_diameter = 6.5 * 2.54  # 6.5" wheel diameter
     running_radius = 0.5 * (
-        2.0 * wheel_diameter / 3.0)  # assume the animal runs at 2/3 the distance from the wheel center
-    running_speed_cm_per_sec = np.pi * speed_deg_per_s * running_radius / 180.
+        # assume the animal runs at 2/3 the distance from the wheel center
+        2.0 * wheel_diameter / 3.0)
+    running_speed_cm_per_sec = angular_speed * running_radius
     return running_speed_cm_per_sec
 
 
-def get_running_df(data, time):
-    dx_raw = data["items"]["behavior"]["encoders"][0]["dx"]
+def _identify_wraps(vsig: Iterable, *,
+                    min_threshold: float = 1.5,
+                    max_threshold: float = 3.5):
+    """
+    Identify "wraps" in the voltage signal. In practice, this is when
+    the encoder voltage signal crosses 5V and wraps to 0V, or
+    vice-versa.
+
+    Argument defaults and implementation suggestion via @dougo
+
+    Parameters
+    ----------
+    vsig: Iterable (array-like)
+        1d array-like iterable of voltage signal
+    min_threshold: float (default=1.5)
+        The min_threshold value that must be crossed to be considered
+        a possible wrapping point.
+    max_threshold: float (default=3.5)
+        The max threshold value that must be crossed to be considered
+        a possible wrapping point.
+
+    Returns
+    -------
+    Tuple
+        Tuple of ([indices of positive wraps], [indices of negative wraps])
+    """
+    # Compare against previous value
+    shifted_vsig = _shift(vsig)
+    if not isinstance(vsig, np.ndarray):
+        vsig = np.array(vsig)
+    # Suppress warnings for when comparing to nan values
+    with np.errstate(invalid='ignore'):
+        pos_wraps = np.asarray(
+            np.logical_and(vsig < min_threshold, shifted_vsig > max_threshold)
+            ).nonzero()[0]
+        neg_wraps = np.asarray(
+            np.logical_and(vsig > max_threshold, shifted_vsig < min_threshold)
+            ).nonzero()[0]
+    return pos_wraps, neg_wraps
+
+
+def _local_boundaries(time, index, span: float = 0.25) -> tuple:
+    """
+    Given a 1d array of monotonically increasing timestamps, and a
+    point in that array (`index`), compute the indices that form the
+    boundary around `index` for timespan `span`.
```

Comment:
I think this docstring should indicate whether the endpoints are inclusive or exclusive.